src/main/java/ui/issuepanel/PanelControl.java

```diff
@@ -39,6 +45,8 @@ public PanelControl(UI ui, Preferences prefs) {
                 setCurrentlySelectedPanel(Optional.of(e.panelIndex)));
         ui.registerEvent((PanelClickedEventHandler) e ->
                 setCurrentlySelectedPanel(Optional.of(e.panelIndex)));
+        ui.registerEvent((ShowRenamePanelEventHandler) e -> Platform.runLater(() ->
+                ((FilterPanel) getPanel(e.panelId)).showRenameTextField()));
```

Comment:
Yes, but `ShowRenamePanelEvent` is only triggered via clicks, from the UI thread. Removing `Platform.runLater` should be fine.

Which event handler are you referring to?